### ISSUE TITLE
Using minimalist for demos

### ISSUE THREAD

**Issue by Rumyra** [mdn/mdn-minimalist#384]
This isn't so much an issue as a way to open a discussion.

I write docs for MDN and also create small demos to support the docs I'm writing (see https://github.com/mdn/dom-examples for some examples).

I wrote a very tiny css file based on MDN styling a while ago, which I include when creating demos to easily style things: https://github.com/mdn/houdini-examples/blob/master/sharedAssets/shared-styles.css

What would be great is if I could substitute that _or_ create something similar but using minimalist instead.

My question is do you have any plans to include a generated css file for inclusion on smaller projects? I avoid tooling when building demos for docs (as do most).

Thanks!

**Comment by schalkneethling** [comment]
Hey @Rumyra - Thanks so much for your inquiry. That is exactly what this is made for :) Let me think about the best way to do that and I will let you know. If you have any suggestions, I am all ears.

**Comment by mattiapontonio** [comment]
If you want minimalist to have an effect on content, can't you put the html in content (instead of in the other repository that is embedded)?

**Comment by Rumyra** [comment]
That's precisely what I'm suggesting @mattiapontonio - some way to easily link to a compiled version of minimalist. These are external standalone demos, that we either link to or embed on MDN, most commonly for tutorials or how tos.

@schalkneethling  I was looking at git hooks for a personal project which is pretty similar. A standalone base styling that I use on a variety of different projects. I didn't get too far however as it seemed particularly complicated to just run one sass build command before commit - I still believe this to be a good solution I just haven't investigated further yet.

**Comment by mattiapontonio** [comment]
Even Just compiled and not minified 👐

**Comment by schalkneethling** [comment]
> @schalkneethling I was looking at git hooks for a personal project which is pretty similar. A standalone base styling that I use on a variety of different projects. I didn't get too far however as it seemed particularly complicated to just run one sass build command before commit - I still believe this to be a good solution I just haven't investigated further yet.

This ability is coming this week :)

**Comment by Rumyra** [comment]
Yay!! That's awesome @schalkneethling - I'm working on a demo repo at the mo that I would love to try this in \o/

Thanks :)

**Comment by schalkneethling** [comment]
@Rumyra please have a look and let me know if this works for you:
https://github.com/mdn/mdn-minimalist/tree/main/dist/css

**Comment by schalkneethling** [comment]
@Rumyra I am wondering whether it would be better if the fonts were just included/embedded in the CSS? I know it is not great/ideal to have them as say base64 but, it might resolve some possible problems with fonts that fail to load.

**Comment by schalkneethling** [comment]
@Rumyra actually, I might have an even better idea 😄

**Comment by schalkneethling** [comment]
@Rumyra essentially I will write a small Node script that will allow you to do something like:

```
npx @mdn/minimalist css/lib
```

This will then copy the CSS and all relevant assets into the above folder from the repo. Once there is an update, you simply run the above command again.

**Comment by Rumyra** [comment]
This sounds like a brilliant idea @schalkneethling - I just tried adding the css into a current project and it doesn't like the github raw link - which I'm still trying to figure out.

However if I copy out the compiled css and make a local file from it it works _very_ well. Albeit the fonts, which I agree possibly quite a load to base64 them but that is a possible solution. Otherwise maybe I just grab them for each time I use minimalist.

Of course I'm happy to do this, but I guess the question is would others?

**Comment by mattiapontonio** [comment]
Reading this fast I am told that the repository that wants to use minimalist must have minimalist as a dependency assuming it is in Node with a package.json

https://docs.npmjs.com/cli/v6/configuring-npm/package-json

But I keep voting for a compiled, versioned and serviced version of minimalist because I like to read compiled CSS.

**Comment by schalkneethling** [comment]
@Rumyra yeah, I thought that might be the case. I reckon that script is going to be the glue this needs. I will let you know once you can test it out.

**Comment by schalkneethling** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Using minimalist for demos](https://trello.com/c/UqfM9HjZ/25-using-minimalist-for-demos)

**Comment by schalkneethling** [comment]
So the action to build the CSS on merges to master totally works!!
https://github.com/mdn/mdn-minimalist/commit/363772663ed238c21df068cbb6ff78b3cc55afd6

**Comment by imme-emosol** [comment]
Only slightly related, but more of a question than an issue, to get something like https://mdn-minimalist.netlify.app/ locally, do I need to build from mdn/yari ? Reading a bit inbetween the lines, #64 seems to hint at that.
An issue could be to add this info to the readme?

**Comment by schalkneethling** [comment]
> An issue could be to add this info to the readme?

Oh boy yes 😸  The README needs some love.

To get this locally, you would run:

```
yarn build:styleguide
```

https://github.com/mdn/mdn-minimalist/blob/0633fb9acae56e5c5b9f14fde683907ded593159/package.json#L9
